Cargo.toml

```diff
@@ -1,45 +1,54 @@
 [package]
-name = "sframe"
-version = "0.2.0"
 edition = "2021"
 license = "MIT/Apache-2.0"
+name = "sframe"
+version = "0.2.0"
 
 authors = [
   "Tobias Waurick <tobias.waurick@goto.com>",
   "Hendrik Sollich <hendrik.sollich@goto.com>",
   "Richard Haehne <richard.haehne@goto.com>",
 ]
 description = "pure rust implementation of SFrame draft-ietf-sframe-enc-00"
-repository = "https://github.com/goto-opensource/sframe-rs"
 documentation = "https://docs.rs/sframe/"
 readme = "README.md"
+repository = "https://github.com/goto-opensource/sframe-rs"
 
 keywords = ["encryption", "sframe", "secure-frame", "webrtc"]
 
 [dependencies]
 bitfield = "0.14"
+cfg-if = "1.0"
 log = "0.4"
 thiserror = "1.0"
 
 [dependencies.ring]
+optional = true
 version = "0.16"
 
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+ring = {version = "0.16", features = ["wasm32_c"], optional = true}
+
+[dependencies.openssl]
+optional = true
+version = "0.10"
+
 [dev-dependencies]
-criterion = { version= "0.4", features=["html_reports"] }
+criterion = {version = "0.4", features = ["html_reports"]}
 hex = "0.4"
+lazy_static = "1.4.0"
+phf = {version = "0.11", features = ["macros"]}
 pretty_assertions = "1.3"
 rand = "0.8"
+serde = {version = "1.0", features = ["derive"]}
 serde_json = "1.0"
-serde = { version = "1.0", features = ["derive"] }
-lazy_static = "1.4.0"
 strum_macros = "0.24"
-phf = { version = "0.11", features = ["macros"] }
-
 
 [features]
-default = []
-wasm-bindgen = ["ring/wasm32_c"]
+default = ["ring"]
```

Comment:
Not sure if it is a good habit to use the default feature.
When a user wants to use `openssl`, he has to explicitly use `no-default-features`. 
Maybe it is better to demand that either `ring` or `openssl` has to be supplied explicitly`?

Reply:
do they have to be mutually exclusive?

Reply:
Not necessarly.
I also liked the idea to fallback to the `ring` implementation when both features are active
 However you will have an unnecessary dependency when compiling. Could also be ok, but it would be nice if one could at least print a warning during compilation.  I am not sure if that is possible though.